desktop-exporter/app/routes/trace-view.tsx

```diff
@@ -59,3 +64,25 @@ export default function TraceView() {
     </Grid>
   );
 }
+
+function orderSpans(
+  parentNode: TreeItem,
+  depth: number,
+  orderedSpans: SpanData[],
+) {
+  let parentSpan = parentNode.data;
+  parentSpan.depth = depth;
+  orderedSpans.push(parentSpan);
+
+  let children = parentNode.children.sort(
+    // Not sure if this is how you write this in typescript, but I did my best
+    (a: { data: { startTime: string } }, b: { data: { startTime: string } }) =>
+      getNsFromString(a.data.startTime) - getNsFromString(b.data.startTime),
+  );
+
+  children.forEach((node: TreeItem) => {
+    orderedSpans = orderSpans(node, depth + 1, orderedSpans);
+  });
+
+  return orderedSpans;
+}
```

Comment:
I think this might be clearer written without recursion, using a stack, and likely easier to handle the orphans.

This is a quick sketch:

```typescript
for (let root of spanTree) {
  let stack = [{
    node: root.data,
    depth: 0,
  }];

  while (stack.length) {
    let {node, depth} = stack.pop();

    orderedSpans.push({
      span: node,
      depth: depth
    })

    node.children
      .sort(...)
      .forEach(child => stack.push({ node: child.data, depth: depth + 1}));
  }
}
```

Reply:
I did it! 😁